_query-dsl/specialized/script-score.md

```diff
@@ -328,5 +328,52 @@ GET blogs/_search
 ```
 {% include copy-curl.html %}
 
+### Term frequency functions
+
+Term frequency functions expose term-level statistics in a script score context. You can use these statistics to implement custom information retrieval and ranking algorithms like query-time multiplicative or additive score boosting by popularity. To apply a term frequency function, call one of the following Painless methods:
+
+- `int termFreq(String <field-name>, String <term>)`: Retrieves the term frequency within a field for a specific term.
+- `float tf(String <field-name>, String <term>)`: Calculates the term frequency/inverse document frequency (TF/IDF) for a specific term within a field.
```

Comment:
There's some setup for using `tf`, without them you will get errors like this:

```
{
  "error": {
    "root_cause": [
      {
        "type": "unsupported_operation_exception",
        "reason": "unsupported_operation_exception: requires a TFIDFSimilarity (such as ClassicSimilarity)"
      }
    ],
    "type": "search_phase_execution_exception",
    "reason": "all shards failed",
    "phase": "query",
    "grouped": true,
    "failed_shards": [
      {
        "shard": 0,
        "index": "test",
        "node": "sQkgvgQWRT2pfI4leTC8lg",
        "reason": {
          "type": "script_exception",
          "reason": "runtime error",
          "script_stack": [
            "org.opensearch.ExceptionsHelper.convertToOpenSearchException(ExceptionsHelper.java:86)",
            "org.opensearch.script.ScoreScriptUtils$TF.tf(ScoreScriptUtils.java:113)",
            "tf(params.field, params.term)",
            "                       ^---- HERE"
          ],
          "script": "tf(params.field, params.term)",
          "lang": "painless",
          "position": {
            "offset": 23,
            "start": 0,
            "end": 29
          },
          "caused_by": {
            "type": "exception",
            "reason": "java.lang.UnsupportedOperationException: requires a TFIDFSimilarity (such as ClassicSimilarity)",
            "caused_by": {
              "type": "unsupported_operation_exception",
              "reason": "unsupported_operation_exception: requires a TFIDFSimilarity (such as ClassicSimilarity)"
            }
          }
        }
      }
    ],
    "caused_by": {
      "type": "unsupported_operation_exception",
      "reason": "unsupported_operation_exception: requires a TFIDFSimilarity (such as ClassicSimilarity)"
    }
  },
  "status": 400
}
```

Here's some suggestions.

1. Setup the Similarity Model:
You need to set the similarity model of the field you're working with Ex: ClassicSimilarity, TFIDFSimilarity). Make sure the model is supported under current OpenSearch Version

To do this, you can update the mappings of your index:

```
PUT /test/_mapping
{
  "properties": {
    "<field_name>": {
      "type": "text",
      "similarity": "classic"
    }
  }
}
```

2. Reindex your data for existing indices

Reply:
When I tried to set similarity to "classic", I got an error that suggests to use "BM25" as similarity: 

```
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "The [classic] similarity may not be used anymore. Please use the [BM25] similarity or build a custom [scripted] similarity instead."
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "The [classic] similarity may not be used anymore. Please use the [BM25] similarity or build a custom [scripted] similarity instead."
  },
  "status": 400
}
```

Reply:
Good catch, would you create an issue in OpenSearch repo or I can do it too.

Reply:
I can create an issue

Reply:
Done. The OpenSearch issue is https://github.com/opensearch-project/OpenSearch/issues/9958